crypto/rand/randfile.c

```diff
@@ -83,26 +83,38 @@ int RAND_load_file(const char *file, long bytes)
         return 0;
 
 #ifndef OPENSSL_NO_POSIX_IO
-    if (stat(file, &sb) < 0 || !S_ISREG(sb.st_mode)) {
-        RANDerr(RAND_F_RAND_LOAD_FILE, RAND_R_NOT_A_REGULAR_FILE);
-        ERR_add_error_data(2, "Filename=", file);
-        return -1;
-    }
+    /*
+     * If not a regular file, assume it's a device and don't try to
+     * read everything.
+     */
+    if (stat(file, &sb) >= 0 && !S_ISREG(sb.st_mode) && bytes < 0)
```

Comment:
Formally speaking stat followed by openssl_fopen is error prone, because openssl_fopen is free to do some magic (like de-UTF8-ing the file name on Windows). So that openssl_fopen followed by fstat is the one to use.

Reply:
Just in case, 1.1.0 gets it right, so just follow the lead...